File: ChessApp.Server/Models/Move.cs
namespace ChessApp.Server.Models
{
    public class Move
    {
        public required Piece Piece { get; set; }
        public required Coordinate From { get; set; }
        public required Coordinate To { get; set; }
        public required bool IsPromotion {  get; set; }
        public required bool IsCastle { get; set;}
        public required bool IsEnPassant { get; set; }
        public Piece? TakenPiece { get; set; }
        public required string Notation { get; set; }
    }
}

File: chessApp.client/src/types/move.ts
import { Coordinate } from "./Coordinate";
import { Piece } from "./Piece";

export type Move = {
    piece: Piece,
    from: Coordinate;
    to: Coordinate;
    isPromotion: boolean;
    isCastle: boolean;
    isEnPassant: boolean;
    takenPiece: Piece | null;
    notation: string | null;
};